src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/json/ReadJSONStepExecution.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016 CloudBees Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.utility.steps.json;
+
+import java.io.PrintStream;
+import java.io.StringReader;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+
+import hudson.FilePath;
+import hudson.model.TaskListener;
+import net.sf.json.JSON;
+import net.sf.json.JSONObject;
+import net.sf.json.groovy.JsonSlurper;
+
+/**
+ * Execution of {@link ReadJSONStep}.
+ *
+ * @author Nikolas Falco
+ */
+public class ReadJSONStepExecution extends AbstractSynchronousNonBlockingStepExecution<JSON> {
+
+    private static final long serialVersionUID = 1L;
+
+    @StepContextParameter
+    private transient TaskListener listener;
+
+    @StepContextParameter
+    private transient FilePath ws;
+
+    @Inject
+    private transient ReadJSONStep step;
+
+    @Override
+    protected JSON run() throws Exception {
+        PrintStream logger = listener.getLogger();
+        JSON json = new JSONObject();
+
+        JsonSlurper jsonSlurper = new JsonSlurper();
```

Comment:
because I do not know if json is a json array or a json object.
By default I setup json as empty json object `{}` than JsonSlurper instantiates the correct JSONObject or JSONArray.
JsonSluper provides a default jsonConfig and parse from inputstream.